File: backend/internal/domain/overtime.go
package domain

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type OvertimeStatus string

const (
	OvertimePending  OvertimeStatus = "pending"
	OvertimeApproved OvertimeStatus = "approved"
	OvertimeRejected OvertimeStatus = "rejected"
)

type Overtime struct {
	ID              uuid.UUID      `gorm:"type:uuid;primary_key;default:gen_random_uuid()" json:"id"`
	EmployeeID      uuid.UUID      `gorm:"type:uuid;not null;index" json:"employee_id"`
	Date            time.Time      `gorm:"type:date;not null" json:"date"`
	StartTime       time.Time      `gorm:"not null" json:"start_time"`
	EndTime         time.Time      `gorm:"not null" json:"end_time"`
	DurationMinutes int            `gorm:"not null" json:"duration_minutes"`
	Type            string         `gorm:"type:varchar(50);default:'working_day'" json:"type"`
	Reason          string         `gorm:"type:text;not null" json:"reason"`
	Status          OvertimeStatus `gorm:"type:varchar(20);default:'pending'" json:"status"`
	ApprovedBy      *uuid.UUID     `gorm:"type:uuid" json:"approved_by,omitempty"`
	RejectReason    *string        `gorm:"type:text" json:"reject_reason,omitempty"`
	CreatedAt       time.Time      `gorm:"default:now()" json:"created_at"`

	Employee *Employee `gorm:"foreignKey:EmployeeID;constraint:OnDelete:CASCADE;"`
	Approver *Employee `gorm:"foreignKey:ApprovedBy;constraint:OnDelete:CASCADE;"`
}

func (o *Overtime) BeforeCreate(tx *gorm.DB) (err error) {
	if o.ID == uuid.Nil {
		o.ID = uuid.New()
	}
	return
}

File: mobile/lib/features/overtime/domain/entities/overtime.dart
import 'package:equatable/equatable.dart';

class Overtime extends Equatable {
  final String id;
  final String employeeId;
  final DateTime date;
  final DateTime startTime;
  final DateTime endTime;
  final int durationMinutes;
  final String type;
  final String reason;
  final String status;
  final String? rejectReason;

  const Overtime({
    required this.id,
    required this.employeeId,
    required this.date,
    required this.startTime,
    required this.endTime,
    required this.durationMinutes,
    required this.type,
    required this.reason,
    required this.status,
    this.rejectReason,
  });

  factory Overtime.fromJson(Map<String, dynamic> json) {
    return Overtime(
      id: json['id'] ?? '',
      employeeId: json['employee_id'] ?? '',
      date: DateTime.parse(json['date']),
      startTime: DateTime.parse(json['start_time']),
      endTime: DateTime.parse(json['end_time']),
      durationMinutes: json['duration_minutes'] ?? 0,
      type: json['type'] ?? 'working_day',
      reason: json['reason'] ?? '',
      status: json['status'] ?? 'pending',
      rejectReason: json['reject_reason'],
    );
  }

  @override
  List<Object?> get props => [id, employeeId, date, startTime, endTime, durationMinutes, type, reason, status, rejectReason];
}
